test/unit/lib/GitHub.test.js

```diff
@@ -100,25 +100,57 @@ describe('GitHub interface', () => {
 
       const githubInstance = await new GitHub(req.params)
 
+      let exception = 'thrown'
       try {
         await githubInstance.readFile(filePath)
+        exception = 'not thrown'
       } catch (err) {
         expect(err._smErrorCode).toEqual('GITHUB_READING_FILE')
       }
-
+      expect(exception).toBe('thrown')
       expect(scope.isDone()).toBe(true)
     })
 
-    test('returns an error if parsing fails for the given file', async () => {
+    test('returns an error if the config file cannot be read', async () => {
       const filePath = 'path/to/file.yml'
       const githubInstance = await new GitHub(req.params)
 
+      let exception = 'thrown'
       try {
         await githubInstance.readFile(filePath)
+        exception = 'not thrown'
+      } catch (err) {
+        expect(err._smErrorCode).toEqual('GITHUB_READING_FILE')
+        expect(err.message).toBeDefined()
+      }
+
+      expect(exception).toBe('thrown')
+    })
+
+    test('returns an error if the config file cannot be parsed', async () => {
```

Comment:
I also added a test which *does* test parsing failure: this time the file exists but is invalid YAML.